### PR TITLE
Multiple textbooks

### PR DESCRIPTION
**Pull request by SeanPhilippi**
tested in Robo 3T, works

```js
db.getCollection('courses').find({}).forEach(course => {
  if (course.textbook !== null) {
    db.getCollection('courses').update(
      { "_id": course._id },
      { $set: { "textbooks": [course.textbook] }}
    )
  }
})
```

### CODE REVIEW

File: src/js/components/CourseModalComponent.js [hunk 1]
```diff
@@ -75,8 +69,11 @@ module.exports = React.createBackboneClass({
     this.state.course[e.currentTarget.getAttribute('id')] = e.currentTarget.value;
   },
 
-  changeTextbookValue(e) {
-    this.state.course.textbook = this.props.textbooks.get(e.currentTarget.value);
+  selectTextbooks(options) {
+    this.setState({ textbooks: options });
+    this.state.course.textbooks = options.map(textbook => {
```
Comment (stuartkilgore) [inline src/js/components/CourseModalComponent.js]: Shouldn't be setting state directly, only by using `this.setState`.

Comment (kevincolten) [reply src/js/components/CourseModalComponent.js]: Yes, always set with a new value

```js
this.setState({
  textbooks: options,
  course: {
    ...this.state.course,
    textbooks: options.map(textbook => textbook.value)
  }  
})
```

Comment (kevincolten) [reply src/js/components/CourseModalComponent.js]: Yes, always set with a new value

```js
this.setState({
  textbooks: options,
  course: {
    ...this.state.course,
    textbooks: options.map(textbook => textbook.value)
  }  
})
```

File: src/js/components/CourseModalComponent.js [hunk 2]
```diff
@@ -138,13 +135,25 @@ module.exports = React.createBackboneClass({
...
+    })
+
     this.setState({
       title: nextProps.title,
       course: this.getModel().attributes,
       term: this.getModel().get('term'),
       location: this.getModel().get('location'),
-      textbook: this.getModel().get('textbook'),
+      textbooks: textbookOptions.filter(textbook => {
+        console.log(this.getModel().get('textbooks').map(e => e.id))
```
Comment (kevincolten) [inline src/js/components/CourseModalComponent.js]: remove debugging console.logs

File: src/js/components/CourseModalComponent.js [hunk 3]
```diff
@@ -179,9 +188,9 @@ module.exports = React.createBackboneClass({
       };
     });
 
-    const textbookOptions = this.props.textbooks.map(textbook => {
-      return (<option key={textbook.id} value={textbook.id}>{textbook.get('name')}</option>)
-    });
+    // const textbookOptions = this.props.textbooks.map(textbook => {
+    //   return (<option key={textbook.id} value={textbook.id}>{textbook.get('name')}</option>)
+    // });
```
Comment (kevincolten) [inline src/js/components/CourseModalComponent.js]: you can just delete old code

File: src/js/models/TextbookModel.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
-import { Model } from 'backbone';
+import * as Backbone from 'backbone';
```
Comment (kevincolten) [inline src/js/models/TextbookModel.js]: we shouldn't have needed to do this

File: src/js/components/CourseModalComponent.js [hunk 5]
```diff
@@ -138,13 +135,25 @@ module.exports = React.createBackboneClass({
...
       };
     })
-      .filter(e => e.user.attributes.is_instructor === true);
+    .filter(e => e.user.attributes.is_instructor === true);
+
+    const textbookOptions = this.props.textbooks.models.map(textbook => {
+      return {
+        value: textbook.id,
+        label: textbook.get('name'),
+        textbook: textbook
```
Comment (stuartkilgore) [inline src/js/components/CourseModalComponent.js]: `textbook: textbook` -> `textbook,`

File: src/js/components/CourseModalComponent.js (hunk 2 above)
Comment (stuartkilgore) [inline src/js/components/CourseModalComponent.js]: Forgot a log?

File: src/js/components/CourseModalComponent.js [hunk 6]
```diff
@@ -179,9 +188,9 @@ module.exports = React.createBackboneClass({
       };
     });
 
-    const textbookOptions = this.props.textbooks.map(textbook => {
-      return (<option key={textbook.id} value={textbook.id}>{textbook.get('name')}</option>)
-    });
+    // const textbookOptions = this.props.textbooks.map(textbook => {
```
Comment (stuartkilgore) [inline src/js/components/CourseModalComponent.js]: Shouldn't need to comment out lines to remove something as github is a complete record of the code changes.

File: src/js/components/CourseModalComponent.js [hunk 7]
```diff
@@ -210,17 +219,17 @@ module.exports = React.createBackboneClass({
...
-              <ControlLabel>Textbook</ControlLabel>
+            <FormGroup controlId="textbooks">
+              <ControlLabel>Textbooks</ControlLabel>
               <Select
                 name="textbooks"
-                value={this.state.textbook.id}
-                options={this.props.textbooks.map(textbook => { return { value: textbook.id, label: textbook.get('name')}; })}
-                onChange={this.setTextbookValue}
-                placeholder="Type to search..."
+                value={this.state.textbooks}
```
Comment (stuartkilgore) [inline src/js/components/CourseModalComponent.js]: Using `this.state` a lot, consider deconstruction.

File: src/js/components/TextbooksListComponent.js [hunk 8]
```diff
@@ -25,7 +26,8 @@ module.exports = React.createBackboneClass({
     this.state.textbook.clear().set(textbook.attributes);
     this.setState({
       showModal: true,
-      modalTitle: textbook.id ? 'Edit Textbook' : 'New Textbook'
+      modalTitle: textbook.id ? 'Edit Textbook' : 'New Textbook',
+      deleteBtn: textbook.id ? true : false
```
Comment (stuartkilgore) [inline src/js/components/TextbooksListComponent.js]: `deleteBtn: !!textbook.id`

**Pull request merged**